### ISSUE TITLE
Add Dartium to the path on package bots

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#23816]
Currently, although Dartium is installed on the package bots, the `dartium` executable doesn't seem to be on the path. Since there's no standard installation directory for Dartium, the test runner relies on path lookup to run it.

**Comment by whesse** [comment]
There are two possible ways to put dartium on the bots, and on the path.  All the other browsers have a single installation on the actual build slave, which is on the path, and which is used by all buildbot runs.

The current way dartium is downloaded by test.py when it runs with -r dartium, is that it is downloaded from the google storage bucket dartium-archive, which has the latest _bleeding-edge_ dartium that was built.  I think this is not what we want, considering that we only use the latest dev branch release for the SDK.

So the two alternatives are to install a single dartium on the machine, outside the build directories, and put it on the path.  This is easy, but automatic updating of it is hard.

Otherwise, we want a copy of dartium in each build directory, which means about 50 copies of dartium on each machine.  Then we want a system for updating them to the latest dev channel build, the same way the sdk is downloaded.  But the sdk is downloaded fresh for each run, even if it hasn't changed, so we would want to avoid this for the dartium download.  Then we also need to put this download's directory (different for each buildbot run) into the environment, and make it available to later build steps.  This is much trickier.

So I propose installing dartium on the path of each bot, for now.  How often do you need dartium to be updated?

**Comment by nex3** [comment]
> So I propose installing dartium on the path of each bot, for now. How often do you need dartium to be updated?

Although for the most part we don't need bleeding-edge features in Dartium, we are going to be doing more with the VM service protocol which may require us to run tests against a fairly recent version. Bleeding edge would be ideal, but we could make do with the latest dev release.

**Comment by ricowind** [comment]
DBC: There are benefits of both dev and bleeding edge. Main downside of just pulling TOT bleeding edge builds (that are green) is that you get an unpredictable extra vertical. If a test is failing, you need to consider the possibility that there is something wrong/changed with dartium on bleeding edge. If you don't surface that information, people may be looking in the wrong spot for a failure for a long time.
So in short, if we use dartium bleeding edge we should make sure people get that information when a test is failing on dartium (this may be as simple as "PLEASE note that this is running on version X.Y.Z of dartium from bleeding edge, get the corresponding copy from https://storage.google.com/.... and put it in foo/bar/drt locally to reproduce)

**Comment by whesse** [comment]
Dartium no longer has an executable named dartium.exe or dartium.  The executable is named chrome.  So the strategy used in package:test to find it will not work.  Package:test also has other outdated code, about finding dartium in the editor checkout.  I think the whole strategy of package:test for finding dartium needs to be redone, perhaps with an environment variable to specify the path or the executable.

I did put the current dev channel dartium on the path of the Windows bot, with name dartium, so you can see how that works.  We can do this on the other platforms, without a problem, if that is what you want.

**Comment by nex3** [comment]
> I think the whole strategy of package:test for finding dartium needs to be redone, perhaps with an environment variable to specify the path or the executable.

The current approach seems to be working well for users. In practice, no one actually uses the executable name `chrome` to refer to Dartium, because that would be confusing and potentially clobber use of *actual* Chrome. We certainly can't just look for `chrome` and assume it's Dartium.

Given that users need to explicitly specify where their Dartium executable is, creating a symlink or wrapper script called `dartium` is a good way of doing so. Unlike an environment variable, it's something that users are likely to have set up already (for example, Homebrew does it automatically) and it's useful for manual invocations as well as using the test runner.

**Comment by athomas** [comment]
Dartium will no longer be supported on 2.0. Closing.
